actors/miner/src/expiration_queue.rs

```diff
@@ -114,12 +143,18 @@ impl ExpirationSet {
     /// A set is empty if it has no sectors.
     /// The power and pledge are not checked, but are expected to be zero.
     pub fn is_empty(&self) -> bool {
-        self.on_time_sectors.is_empty() && self.early_sectors.is_empty()
+        self.on_time_sectors.is_empty()
+            && self.proof_expiring_sectors.is_empty()
+            && self.faulty_sectors.is_empty()
     }
 
     /// Counts all sectors in the expiration set.
     pub fn len(&self) -> u64 {
-        self.on_time_sectors.len() + self.early_sectors.len()
+        self.on_time_sectors.len() + self.proof_expiring_sectors.len() + self.faulty_sectors.len()
```

Comment:
Is it an invariant that these proof_expiring_sectors does not overlap with on_time_sectors?  I don't think it is: what if someone refreshes their proof beyond current sector expiration and then extends sector life time to fit in the same quantized epoch? If that's right then we undercount and instead we should use `self.all().len()`

Reply:
When adding a sector into the expiration queue or moving it around, we first decide if the expiration is going to be proof expiration or commitment expiration and then add it to the corresponding queue. If the above invariant is broken, there are mistakes in the code.